### PR TITLE
threading related deprecation fixes for python 3.10

### LINKED ISSUE #1838: Threading related deprecations in Python 3.10
notifyAll has been deprecated in favor of notifyAll
currentThread has been deprecated in favor of current_thread
setDaemon has been deprecated in favor of setting daemon attribute

Reference : https://bugs.python.org/issue43723

```
➜  paramiko git:(master) rg -t py -w 'currentThread|notifyAll|activeCount|getName|setName|isDaemon|setDaemon'
demos/rforward.py
83:        thr.setDaemon(True)

paramiko/buffered_pipe.py
104:            self._cv.notifyAll()
206:            self._cv.notifyAll()

paramiko/util.py
235:    tid = id(threading.currentThread())

paramiko/sftp_file.py
530:        t.setDaemon(True)

paramiko/transport.py
420:        self.setDaemon(True)

paramiko/channel.py
1069:            self.out_buffer_cv.notifyAll()
1233:        self.out_buffer_cv.notifyAll()

tests/loop.py
84:            self.__cv.notifyAll()

tests/test_transport.py
747:                self.setDaemon(True)
771:                self.setDaemon(True)
```

### PR DESCRIPTION
**Pull request by graingert**
fixes #1838 

remake of https://github.com/paramiko/paramiko/pull/1870 with py2.6 support

### CODE REVIEW

File: demos/rforward.py [hunk 1]
```diff
@@ -80,7 +81,7 @@ def reverse_forward_tunnel(server_port, remote_host, remote_port, transport):
         thr = threading.Thread(
             target=handler, args=(chan, remote_host, remote_port)
         )
-        thr.setDaemon(True)
+        set_daemon(thr, True)
```
Comment (jun66j5) [inline demos/rforward.py:84]: We could use simply `thr.daemon = True` with Python 2.7 and 3.

* https://docs.python.org/2/library/threading.html#threading.Thread.daemon
* https://docs.python.org/3/library/threading.html#threading.Thread.daemon

File: paramiko/buffered_pipe.py [hunk 2]
```diff
@@ -101,7 +101,7 @@ def feed(self, data):
             if self._event is not None:
                 self._event.set()
             self._buffer_frombytes(b(data))
-            self._cv.notifyAll()
+            notify_all(self._cv)
```
Comment (jun66j5) [inline paramiko/buffered_pipe.py:104]: We could use simply `self._cv.notify_all()` with Python 2.7 and 3.

* https://docs.python.org/2/library/threading.html#threading.Condition.notify_all
* https://docs.python.org/3/library/threading.html#threading.Condition.notify_all

File: demos/rforward.py (hunk 1 above)
Comment (graingert) [reply demos/rforward.py:84]: Woops I got the description wrong, this is for py2.6 support

Comment (jun66j5) [reply demos/rforward.py:84]: The `daemon` attribute is available since Python 2.6 and works.

**Comment by graingert** [comment]
These methods have been supported since 2.6

File: demos/rforward.py (hunk 1 above)
Comment (jun66j5) [reply demos/rforward.py:84]: I've confirmed with Python 2.6 and just confirmed now.

```
$ python2.6
Python 2.6.9 (default, Jul 29 2017, 12:47:14)
[GCC 5.4.0 20160609] on linux4
Type "help", "copyright", "credits" or "license" for more information.
>>> import threading
>>> import time
>>> t = threading.Thread(target=lambda: time.sleep(1000))
>>> t.daemon = True
>>> t.start()
>>> t.is_alive()
True
>>> #=> it is able to exit without no wait
```

Comment (jun66j5) [reply demos/rforward.py:84]: > Wait only 3.3 needs the old method names? It seems the new way is supported from 2.6+? [#1870 (comment)](https://github.com/paramiko/paramiko/pull/1870#issuecomment-1063258752)

I recommend checking documentation. The `daemon` and `notify_all` are available on Python 3.0.

* https://docs.python.org/3.0/library/threading.html#threading.Thread.daemon
* https://docs.python.org/3.0/library/threading.html#threading.Condition.notify_all

**Pull request closed without merging**